File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    var answer = 1;
    
    function confirm(start){
        let num = start;
        let sum = 0;
        while(sum < n){
            sum += num;
            if(sum > n) return false;
            else{
                if(sum === n) return true;
                num++;
            }
        }
    }
    for(let i = 1; i <= parseInt(n / 2); i++){
        if(confirm(i)){
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    start_num = 1
    sum_val = 0
    
    for start_num in range(1, n + 1):
        sum_val = 0
        for i in range(start_num, n + 1):
            sum_val += i
            if sum_val == n:
                answer += 1
                break
            elif sum_val > n:
                break
    
    return answer
